task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -174,6 +166,20 @@ def get_template_context(self) -> Context:
             }
             context.update(context_from_server)
 
+            if dag_run.data_interval_start and dag_run.data_interval_end:
+                context.update(
+                    {
+                        "data_interval_end": dag_run.data_interval_end,
+                        "data_interval_start": dag_run.data_interval_start,
+                        "prev_data_interval_start_success": lazy_object_proxy.Proxy(
+                            lambda: get_previous_dagrun_success(self.id).data_interval_start
+                        ),
+                        "prev_data_interval_end_success": lazy_object_proxy.Proxy(
+                            lambda: get_previous_dagrun_success(self.id).data_interval_end
+                        ),
```

Comment:
Doesn’t really matter, but these logically should go under the `if logical_date` block instead.

Reply:
updated